projects/kit/components/input-year/input-year.directive.ts

```diff
@@ -0,0 +1,127 @@
+    providers: [
+        tuiAsControl(TuiInputYearDirective),
+        tuiAsTextfieldAccessor(TuiInputYearDirective),
+        tuiValueTransformerFrom(TUI_INPUT_YEAR_OPTIONS),
+    ],
+    hostDirectives: [TuiWithTextfield, MaskitoDirective, TuiItemsHandlersValidator],
+    host: {
+        inputmode: 'numeric',
+        '[value]': 'value()',
+        '[disabled]': 'disabled()',
+        '(click)': 'toggle()',
+        '(input)': 'onInput($event.target.value)',
+    },
+})
+export class TuiInputYearDirective
+    extends TuiControl<number | null>
+    implements TuiTextfieldAccessor<number | null>
+{
+    private readonly options = inject(TUI_INPUT_YEAR_OPTIONS);
+    private readonly open = tuiDropdownOpen();
+    private readonly handlers = inject(TuiItemsHandlersDirective);
+    private readonly textfield = inject(TuiTextfieldComponent);
+
+    protected readonly calendar = tuiInjectAuxiliary<TuiCalendarYear>(
+        (x) => x instanceof TuiCalendarYear,
+    );
+
+    protected readonly icon = tuiTextfieldIconBinding(TUI_INPUT_YEAR_OPTIONS);
+    protected readonly min = signal(this.options.min.year);
+    protected readonly max = signal(this.options.max.year);
+
+    protected readonly mask = tuiMaskito(
+        computed(() => ({
+            ...maskitoNumberOptionsGenerator({
+                min: this.min(),
+                max: this.max(),
+                thousandSeparator: '',
+            }),
+            mask: UP_TO_4_DIGITS_REG,
+        })),
+    );
+
+    protected readonly calendarIn = effect(
+        () => this.processCalendar(),
+        TUI_ALLOW_SIGNAL_WRITES,
+    );
+
+    protected readonly calendarOut = effect((onCleanup) => {
+        const calendar = this.calendar();
+
+        if (calendar) {
+            const subscription = calendar.yearClick.subscribe((year) =>
+                this.textfield.handleOption(year),
+            );
+
+            onCleanup(() => subscription?.unsubscribe());
+        }
+    });
+
+    // TODO(v5): use signal inputs
+    @Input('min')
+    public set minSetter(x: number) {
+        this.min.set(x);
+    }
+
+    // TODO(v5): use signal inputs
+    @Input('max')
+    public set maxSetter(x: number) {
+        this.max.set(x);
+    }
+
+    public setValue(value: number | null): void {
+        this.onChange(value);
+        this.textfield.value.set(value?.toString() ?? '');
+    }
+
+    protected onInput(value: string): void {
+        this.control?.control?.updateValueAndValidity({emitEvent: false});
+        this.onChange(Number(value));
+        this.processCalendar();
+    }
+
+    protected toggle(): void {
+        this.open.update((x) => !x);
+    }
+
+    protected processCalendar(): void {
+        const calendar = this.calendar();
+
+        if (calendar) {
+            calendar.value = this.value();
+            calendar.initialItem = this.value() ?? calendar.initialItem;
+            calendar.disabledItemHandler = this.handlers.disabledItemHandler();
```

Comment:
I wonder if we can just default to inject(TUI_ITEM_HANDLERS).disabledItemHandler() inside of CalendarYear instead